<launch>
    <arg name="debug" default="false"/>
    <arg name="fullscreen" default="true"/>
    <arg name="start_delay" default="0.0" />
    <arg name="rviz" default="true"/>
    <arg name="args" default=""/>
    <arg name="sleep" default="bash -c 'sleep $(arg start_delay); $0 $@' "/>

    <arg unless="$(arg debug)" name="launch-prefix" value="$(arg sleep)"/>
    <arg     if="$(arg debug)" name="launch-prefix" value="$(arg sleep) gdb -ex run --args"/>

    <arg unless="$(arg fullscreen)" name="rviz-fullscreen" value=""/>
    <arg     if="$(arg fullscreen)" name="rviz-fullscreen" value="--fullscreen"/>

    <param name="robot_description" command="$(find xacro)/xacro '$(find franka_description)/robots/panda/panda.urdf.xacro' hand:=true"/>

    <!-- 0. RViz -->
    <node if="$(arg rviz)" type="rviz" name="rviz" pkg="rviz" args="-d $(find smpl_test)/config/config.rviz $(arg rviz-fullscreen)"/>

    <!-- 1. call planner -->
    <node pkg="smpl_test" type="callPlannerFetch" name="call_planner_fetch" launch-prefix="$(arg launch-prefix)" args="$(arg args)">
        <!-- robot model (for planning) -->
        <rosparam param="robot_model">
            kinematics_frame:
                panda_link0
                <!-- base_link -->
            chain_tip_link:
                panda_hand
                <!-- gripper_link -->
        </rosparam>

        <!-- collision checking -->
        <!-- TODO: update -->
        <rosparam command="load" file="$(find sbpl_collision_checking_test)/config/collision_model_franka.yaml"/>

        <!-- planner params -->
        <rosparam command="load" file="$(find smpl_test)/config/franka_arm.yaml"/>
        <param name="planning/mprim_filename" value="$(find smpl_test)/config/franka.mprim"/>
        <param name="planning_problems_directory" value="$(find robowflex_library)/problems/panda/bookshelf_small_panda"/>
        <param name="planning_problem_index" value="1"/>

        <param name="planning_algorithm" value="epase"/>  <!-- options: arastar, epase -->
        <param name="goal_type" value="joints"/>  <!-- options: joints, pose -->
        <param name="num_threads" value="4"/>
        <param name="planning_frame" value="odom_combined"/>
    </node>
</launch>
